src/Uno.UWP/Storage/FileIO.cs

```diff
@@ -0,0 +1,222 @@
+				{
+					try
+					{
+						byte firstByte = binReader.ReadByte();
+						switch (firstByte)
+						{
+							case 0xfe:
+								unicodeEncoding = Streams.UnicodeEncoding.Utf16BE;
+								break;
+							case 0xff:
+								unicodeEncoding = Streams.UnicodeEncoding.Utf16LE;
+								break;
+						}
+					}
+					catch
+					{
+						// probably file is empty
+					}
+
+				}
+			}
+
+			AppendWriteTextAsync(file, contents, unicodeEncoding, append);
+		}
+
+		public static IAsyncAction WriteTextAsync(IStorageFile file, string contents) => AppendWriteTextAsync(file, contents, false).AsAsyncAction();
+
+		public static IAsyncAction AppendTextAsync(IStorageFile file, string contents) => AppendWriteTextAsync(file, contents, true).AsAsyncAction();
+
+		public static IAsyncAction WriteLinesAsync(IStorageFile file, IEnumerable<string> lines)
+			=> WriteTextAsync(file, ConvertLinesToString(lines));
+		public static IAsyncAction AppendLinesAsync(IStorageFile file, IEnumerable<string> lines)
+			=> AppendTextAsync(file, ConvertLinesToString(lines));
+
+
+		private static async Task<string> ReadTextTaskAsync(IStorageFile file, Streams.UnicodeEncoding encoding)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			Encoding encodingForReader = Encoding.UTF8;
+			switch (encoding)
+			{
+				case Streams.UnicodeEncoding.Utf8:
+					encodingForReader = Encoding.UTF8;
+					break;
+				case Streams.UnicodeEncoding.Utf16LE:
+					encodingForReader = Encoding.Unicode;
+					break;
+				case Streams.UnicodeEncoding.Utf16BE:
+					encodingForReader = Encoding.BigEndianUnicode;
+					break;
+			}
+
+			string output = "";
+			using (Stream fileStream = await file.OpenStreamForReadAsync())
+			{
+				using (StreamReader streamReader = new StreamReader(fileStream, encodingForReader))
+				{
+					output = streamReader.ReadToEnd();
+				}
+			}
+			return output;
+		}
+
+		public static IAsyncOperation<string> ReadTextAsync(IStorageFile file, Streams.UnicodeEncoding encoding)
+			=> ReadTextTaskAsync(file, encoding).AsAsyncOperation<string>();
+
+		private static async Task<string> ReadTextTaskAsync(IStorageFile file)
+		{
+			if (file is null)
+			{
+				// UWP throws NullReferenceException
+				// but it was choosen to throw ArgumentNullException , as it is more appriopriate in this caase
+				throw new ArgumentNullException("StorageFile cannot be null");
+			}
+
+			string output = "";
+			Stream fileStream = null;
+			try
+            {
+				fileStream = await file.OpenStreamForReadAsync();
+				using (StreamReader streamReader = new StreamReader(fileStream))
+				{
+					output = streamReader.ReadToEnd();
+				}
+			}
+			finally
+			{
+				if(fileStream != null)
+                {
+					fileStream.Dispose();

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to make sure 'fileStream' is disposed only once.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4827590)

Reply:
But this is already implemented... conversion from using/using to try/using/finally.